package/harvester-os/files/etc/systemd/system/rke2-shutdown.service

```diff
@@ -0,0 +1,14 @@
+[Unit]
+Description=Kill RKE2 Containers for clean unmount
+Documentation=man:systemd-halt.service(8)
+DefaultDependencies=no
+RefuseManualStart=yes
+Before=reboot.target halt.target shutdown.target poweroff.target umount.target final.target
+
+[Install]
+RequiredBy=reboot.target halt.target shutdown.target poweroff.target umount.target final.target
+
+[Service]
+Type=oneshot
+RemainAfterExit=yes
+ExecStart=-/usr/local/bin/rke2-killall.sh
```

Comment:
Also `rke2-killall.sh` sends `KILL (-9)` signals to processes, but rke2-server/agent's systemd services's `ExecStopPost` option sends `TERM` signals. Just wondering if it's safer to use TERM.

Reply:
using TERM might be a good idea to give the processes some time to clean up and terminate.
but we need to have another script to do it?

Reply:
If that's oneliner, we can also embed it in the systemd unit file. There are many cleanups in the rke2-killall.sh script, maybe some are unnecessary.